frontend/src/api/prometheus/__tests__/useQueryRangeResourceData.spec.ts

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios';
+import { act } from '@testing-library/react';
+import { TimeframeTitle } from '~/pages/modelServing/screens/types';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { mockPrometheusServing } from '~/__mocks__/mockPrometheusServing';
+import useQueryRangeResourceData from '~/api/prometheus/useQueryRangeResourceData';
+
+jest.mock('axios', () => ({
+  post: jest.fn(),
+}));
+
+const mockAxios = jest.mocked(axios.post);
+
+describe('useQueryRangeResourceData', () => {
+  const responsePredicate = jest.fn();
+  const query = 'testQuery';
+  const end = 123456;
+  const namespace = 'testNamespace';
+
+  const timeframe: TimeframeTitle = TimeframeTitle.ONE_HOUR;
+
+  it('should fetch data successfully and handle refresh ', async () => {
+    mockAxios.mockResolvedValue({ data: { response: mockPrometheusServing({}) } });
+    responsePredicate.mockReturnValue({ data: { response: mockPrometheusServing({}) } });
+
+    const renderResult = testHook(useQueryRangeResourceData)(
+      true,
+      query,
+      end,
+      timeframe,
+      responsePredicate,
+      namespace,
+    );
+
+    expect(renderResult).hookToStrictEqual({
+      data: [],
+      error: undefined,
+      loaded: false,
+      pending: true,
+      refresh: expect.any(Function),
+    });
+
+    // Wait for next update
+    await renderResult.waitForNextUpdate();
+
+    expect(renderResult).hookToStrictEqual({
+      data: { data: { response: mockPrometheusServing({}) } },
+      error: undefined,
+      loaded: true,
+      pending: false,
+      refresh: expect.any(Function),
+    });
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+
+    // handle refresh
+    mockAxios.mockResolvedValue({ data: { response: mockPrometheusServing({}) } });
+
+    await act(() => renderResult.result.current.refresh());
+
+    expect(mockAxios).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToHaveUpdateCount(3);
+  });
+  it('should handle error when active is false', async () => {
+    testHook(useQueryRangeResourceData)(false, query, end, timeframe, responsePredicate, namespace);
+    expect(mockAxios).toHaveBeenCalledTimes(0);
+  });
+  it('should handle error when request fails', async () => {
```

Comment:
Check `toHaveBeenCalledTimes` and `hookToHaveUpdateCount` before and after the update